libcrux-intrinsics/src/avx2.rs

```diff
@@ -26,7 +26,13 @@ pub fn mm256_storeu_si256_i32(output: &mut [i32], vector: Vec256) {
 }
 
 pub fn mm_storeu_si128(output: &mut [i16], vector: Vec128) {
-    // debug_assert_eq!(output.len(), 8);
+    //debug_assert_eq!(output.len(), 8);
```

Comment:
Do you want to restore this assert?

Reply:
I tried, but I seem to be getting Eurydice-related errors?

Reply:
Removed the assert again for now.

Reply:
You changed the code, hence the C code changed. There's no error in that though.
However, this assert isn't good. We should allo outputs that are longer to avoid unnecessary code on the caller. I did that.

Reply:
Does that also go for the assert in `mm_storeu_si128_i32`, then, @franziskuskiefer?

Reply:
It's a bit of a style question I think. Having clear bounds can be nice (then we can have an assert here and use it in the proofs). But it also requires calling the functions with a slice of the right size (`&x[a...b]`), which in turn generates ugly slices in C. The Rust side is ok, the C side isn't great.
Overall I feel like handling slice only with the required bound (not the exact size) makes everything a little nicer.